### PR TITLE
Move to js

### PR DESCRIPTION
**Pull request by delvedor**
As titled.
I've rewritten all the files in js and removed the coffeescript stuff.
I've also applied the changes that @as-cii did in #40.

This is my first contribution to the Atom project, if I did something wrong, please tell me :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,22 +1,31 @@
 {
   "name": "go-to-line",
   "version": "0.31.2",
-  "main": "./lib/go-to-line-view",
+  "main": "./lib/go-to-line-view.js",
```
Comment (50Wliu) [inline package.json]: I don't think this needs to be changed

**Comment by lee-dohm** [comment]
Thanks for helping out! If you want to help us move to JavaScript, you should take a look at some other packages that we're using JavaScript in, such as [line-ending-selector](https://github.com/atom/line-ending-selector), and model your code after those. In particular, using classes instead of prototypes and using `import` instead of `require`.

Take a look and let us know if you have any questions! Thanks again :grinning:

**Comment by delvedor** [comment]
Your welcome!
Soon the commit with the updates :)

File: spec/go-to-line-spec.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ describe('GoToLine', () => {
       editor = atom.workspace.getActiveTextEditor()
       editorView = atom.views.getView(editor)
       goToLine = GoToLineView.activate()
+      console.log(goToLine)
```
Comment (Ben3eeE) [inline spec/go-to-line-spec.js]: 🔥

Comment (delvedor) [inline spec/go-to-line-spec.js]: Ops! Thanks!
Fixed :)

**Comment by lee-dohm** [comment]
@ungb Could you take a look at this and make sure that everything is 🆗 ?

**Comment by delvedor** [comment]
Guys, some news about this? :)

**Comment by ungb** [comment]
Sorry, will take a look today.  @delvedor Could you resolve the merge conflicts?

**Comment by lee-dohm** [comment]
There's even a new feature to help with merge conflicts: https://github.com/blog/2293-resolve-simple-merge-conflicts-on-github :tada:

**Comment by ungb** [comment]
@lee-dohm I don't think it would work in this case.

image

**Comment by lee-dohm** [comment]
Good point, @ungb. I got carried away :grinning:

**Comment by ungb** [comment]
I was able to test your branch and didn't see any issues from before and after.  We should merge this once the merge conflicts are resolved.

**Comment by delvedor** [comment]
I'm sorry but I am abroad for work, thank you for the merge! :)

**Pull request merged**